test/framework/src/main/java/org/elasticsearch/test/TransportVersionUtils.java

```diff
@@ -14,13 +14,24 @@
 
 import java.util.Collections;
 import java.util.List;
+import java.util.Optional;
 import java.util.Random;
 import java.util.Set;
 import java.util.stream.Collectors;
 
 public class TransportVersionUtils {
     private static final List<TransportVersion> ALL_VERSIONS = List.copyOf(TransportVersion.getAllVersions());
 
+    /** Returns all released versions */
+    public static List<TransportVersion> allReleasedVersions() {
```

Comment:
The version in `TransportVersion` is used to populate the map used for `fromId`, to pre-cache known ids, as well as performing some sanity checking on the class itself.

Reply:
Those additional sanity checks are fine, but then the method is public. Perhaps it could be package private and then move TransportVersionUtils to the same package?

Reply:
Done - added a `TransportVersions` class to access